RELEASE_PROCESS.md

```diff
@@ -127,10 +127,35 @@ based on the branch you just created.
 - [ ] Send an email to the chef and chef-dev mailing lists announcing
 the release with a link to the blog post.
 
-Chef Server is now released.
 
 ## Post Release
 
-- [ ] Bump the Chef Server version number in
-  `omnibus/config/projects/chef-server.rb` for development to the next
-  logical version number.
+Before the release can be considered complete, the following
+post-release steps must be performed;
+
+- [ ] in `omnibus/config/projects/chef-server.rb`:
+  - [ ] Bump the Chef Server version number for development to the next logical
+        version number.
+  - [ ] Update chef client to the latest stable chef-client version via
+        `override :chef`.  This may include updates to major and minor
```

Comment:
Maybe this is implicit, but we should also override ohai.  The default for both of these is to pull from master of chef/chef and chef/ohai and ohai has a circular dependency on the chef repo (to pull chef-config).  Only during chef-client releases is there a good guarantee that all the versions agree without random duplicate gems.